src/client.ts

```diff
@@ -522,44 +596,64 @@ export class Tier {
       req.at = at
     }
     req.clobber = !!clobber
-
     return await this.apiPost<ReportRequest>('/v1/report', req)
   }
 
   public async subscribe(
     org: OrgName,
     features: Features | Features[],
-    effective?: Date,
-    info?: OrgInfo
+    { effective, info, trialDays, checkout }: SubscribeOptions = {}
   ): Promise<{}> {
-    // deprecated overloading of subscribe() as schedule()
-    if (
-      Array.isArray(features) &&
-      !features.some(f => !isPhase(f)) &&
-      effective === undefined &&
-      info === undefined
-    ) {
-      const msg = `Using phase objects with subscribe() is deprecated, please use tier.schedule() for this use case.`
-      process.emitWarning(msg, 'DeprecationWarning')
-      return await this.schedule(org, features as unknown as Phase[])
-    }
-
     const phases: Phase[] = !Array.isArray(features)
       ? [{ features: [features], effective }]
       : features.length
       ? [{ features, effective }]
       : []
-    return await this.schedule(org, phases, info)
+
+    if (trialDays && (typeof trialDays !== 'number' || trialDays <= 0)) {
```

Comment:
Use types?

Reply:
It's technically `number | undefined`, and for some reason TS wasn't picking up the restriction from the truthy check at the start of that conditional.  Can make it more obvious, though.